### DISCUSSION TITLE
关于pyarmor_runtime.so所需要的python环境问题

### DISCUSSION THREAD

**Discussion by yunkai-jia** [dashingsoft/pyarmor#1179] [Q&A]
您好，老师，打扰了

电脑对应的是darwin.x86_64, 针对pyarmor8.1.7加密后的使用的python环境必须要是mac自带的python解释器么，刚刚测试了下发现好像不可以使用conda创建的虚拟环境，但是自带的python解释器的虚拟环境可以使用？谢谢老师

**Comment by jondy** [comment]
你能分享一下你的 conda 创建的虚拟环境吗？我想知道的是在 conda 创建的环境里面 python 解释器的全路径，以及 `otool -L /path/to/python` 的全部输出

**Comment by yunkai-jia** [comment]
好的，老师，您看这样可以么？

![image](https://user-images.githubusercontent.com/64423558/233531323-2d22ae67-49dd-424a-aff7-94ef75ba265b.png)

针对otool -L /path/to/python，我这边没有输出。输出这样可以么？
otool -L `which python`
/Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9:
	/Library/Frameworks/Python.framework/Versions/3.9/Python (compatibility version 3.9.0, current version 3.9.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)

**Comment by yunkai-jia** [comment]
这是目前我的加密文件在配置的conda环境中使用的
![image](https://user-images.githubusercontent.com/64423558/233533388-a6841c35-4594-4d9a-94ed-5e3ffd7790d1.png)

**Comment by yunkai-jia** [comment]
请问老师，是不是哪一步我做的不太对？

**Comment by jondy** [comment]
抱歉，我没有使用过 conda.

**Comment by yunkai-jia** [comment]
嗯嗯，谢谢老师，之前在使用pyarmor7.x的超级加密模式的时候碰到过同样的问题。
我在想有没有一种可能是pyarmor_runtime.so在mac中自带的python解释器中可以去加载，但是在conda中的python解释器无法加载这个pyarmor_runtime.so？ 谢谢老师
